contracts/account/extensions/ERC7579Modules/SocialRecoveryExecutor.sol

```diff
@@ -0,0 +1,475 @@
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {AccountERC7579} from "../AccountERC7579.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @title Implements a Social Recovery Executor module following ERC-7579
+ *
+ * @dev Features:
+ * - Timelocked Guardian-based recovery
+ * - Recovery Execution Scope is limited to reconfiguring installed Validator Modules
+ * - Recovery Reconfiguration can only be performed by the ERC-7579 Account
+ * - Guardian Signatures replay attack protection via EIP-712 typed data signing
+ * - Replay Attack protection includes: chains, accounts, executors, and recovery attempts
+ */
+contract SocialRecoveryExecutor is IERC7579Module, EIP712, ReentrancyGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    bytes32 private constant RECOVERY_TYPEHASH = keccak256("RecoveryMessage(address account,uint256 nonce)");
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                EVENTS & ERRORS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    event ModuleUninstalledReceived(address indexed account, bytes data);
+    event ModuleInstalledReceived(address indexed account, bytes data);
+
+    event RecoveryExecuted(address indexed account, address indexed targetValidatorModule);
+    event RecoveryCancelled(address indexed account);
+    event RecoveryStarted(address indexed account);
+
+    event ThresholdChanged(address indexed account, uint256 indexed threshold);
+    event TimelockChanged(address indexed account, uint256 indexed timelock);
+    event GuardianRemoved(address indexed account, address indexed guardian);
+    event GuardianAdded(address indexed account, address indexed guardian);
+
+    error InvalidTimelock();
+    error InvalidThreshold();
+    error InvalidGuardians();
+    error InvalidGuardian();
+
+    error CannotRemoveGuardian();
+    error GuardianNotFound();
+    error AlreadyGuardian();
+    error MaxGuardians();
+
+    error InvalidInstalledValidatorModule();
+    error InvalidGuardianSignatures();
+    error InvalidRecoveryCallData();
+
+    error InvalidCallType();
+    error ThresholdNotMet();
+
+    error RecoveryAlreadyStarted();
+    error RecoveryNotStarted();
+    error RecoveryNotReady();
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                CONSTANTS & STORAGE
+    //////////////////////////////////////////////////////////////////////////*/
+
+    string public constant NAME = "SocialRecoveryExecutor";
```

Comment:
I think we need to rationalize here. 

Right now there are:
- public contracts (`NAME` and `VERSION` that cannot be overriden)
- external view functions (`name()` and `version()` that are not virtual)
- name and version in EIP712 (set at construction, that can be fetched using `IERC5267.eip712Domain()`

since the first two are NOT required by any module interface, I'd remove them. All that should be set trough constructor params